docs/spec/operators/index.mdx

```diff
@@ -0,0 +1,53 @@
+---
+title: 'Operators'
+sidebar_position: 0
+description: Documentation for Operators
+---
+
+import DocCardList from '@theme/DocCardList';
+
+Operators are the cornerstone of the Inspetkor Gadget architecture. They are
+modular components implementing the same interfaces
+([1](https://pkg.go.dev/github.com/inspektor-gadget/inspektor-gadget@v0.31.0/pkg/operators#DataOperator),
+[2](https://pkg.go.dev/github.com/inspektor-gadget/inspektor-gadget@v0.31.0/pkg/operators#ImageOperator)).
+They are responsible for specific tasks, such as loading eBPF programs,
+enriching events, sorting output, etc. All operators can generate, drop or
+manipulate data and few of them handle the
+[layers](../../oci.md#image-layers-and-media-types) of gadget OCI image.
+
+Operators are arranged in a pipeline according to their priority:
+
+![Opeators Pipeline](./operators-pipeline.svg)
```

Comment:
> The localmanager and kubemanager operators should be after the oci (ebpf, wasm, btf) operators, together with the filter, sorter, etc. Why did you put them at the beginning of the pipeline?

Too fast copy and paste from a previous image created by @flyth. 

> BTW, (not related with this PR), I just realized the filter should have higher priority so that we avoid enriching (localmanager/kubemanger, formatter, uidgidresolver, etc.) events that will be discarded by the filter operator.

What about if we want to filter by enriched data?

Thanks for the priority data. I added it to each operator, perhaps we can put a document with your comment too.

Reply:
> What about if we want to filter by enriched data?
>
Good point. That's why we put it there hahaha.

> Thanks for the priority data. I added it to each operator, perhaps we can put a document with your comment too.
>
Yep. A central point where to see all prios in orde would be very useful. Maybe somehow auto-generated and verified in the CI?

Reply:
Now that I think better. Maybe we should start using always mermaid graph as they can be easily updated by anyone. WDYT?

Reply:
```mermaid
block-beta
  columns 8
  block:group1:2
    columns 3
    space oci space ebpf wasm btfgen
    style oci fill:#ffffff00,stroke:#ffffff00
  end
  block:group2:2
    columns 1
    kubemanager localmanager
  end
  filter
  sorter
  ...
  block:group3:2
    columns 1
    otelmetrics["otel metrics"]
    cli
  end

```

Reply:
```mermaid
flowchart LR
    oci-->manager
    manager-->filter
    filter-->sorter
    sorter-->dotdotdot["..."]
    dotdotdot-->output
    subgraph oci
        ebpf
        wasm
        btfgen
    end
    subgraph manager[" "]
        direction TB
        kubemanager~~~localmanager
    end
    subgraph output[" "]
        otelmetrics["otel metrics"]~~~cli
    end
```

Reply:
With mermaid, we can make boxes clickable to the right section in the documentation.

Reply:
I took the version without the clickable box. I wanted them to point to the operator documentation but I wasn't able.

Reply:
@mauriciovasquezbernal I think you mistakenly merged the version with the links.

Reply:
ops, opened #3437